File: parens.jl
#!/Applications/Julia-0.5.app/Contents/Resources/julia/bin/julia

function _parens(prefix::String, left::Int64, right::Int64)
    left == 0 && right == 0 && println(prefix)
    left > 0 && _parens(string(prefix, '('), left-1, right)
    right > left && _parens(string(prefix, ')'), left, right-1)
end

parens(n::Int64) = _parens("", n, n)

parens(parse(Int64, ARGS[1]))

File: parens.cs
using System;
namespace Catalan
{
  class Benchmark
  {
    static void Parens(string prefix, int left, int right) {
      if (left == 0 && right == 0) {
        Console.WriteLine(prefix);
      }
      if (left > 0) {
        Parens(prefix + "(", left - 1, right);
      }
      if (right > left) {
        Parens(prefix + ")", left, right - 1);
      }
    }
    static void Parens(int n) {
      Parens("", n, n);
    }
    static void Main(string[] args)
    {
      Parens(Int32.Parse(args[0]));
    }
  }
}
